tests/_test_msui/test_mscolab.py

```diff
@@ -901,6 +904,39 @@ def test_profile_dialog(self, qtbot):
             critbox.assert_called_once()
         assert not self.window.mscolab.profile_dialog.gravatarLabel.pixmap().isNull()
 
+    def test_upload_and_fetch_profile_image(self, qtbot):
+        self._connect_to_mscolab(qtbot)
+        self._login(qtbot, self.userdata[0], self.userdata[2])
+        self.window.mscolab.profile_action.trigger()
+
+        initial_pixmap = self.window.mscolab.profile_dialog.gravatarLabel.pixmap().toImage()
+
+        # Creating a new image
+        new_image = Image.new('RGB', (64, 64), color='cyan')
+        img_byte_arr = io.BytesIO()
+        new_image.save(img_byte_arr, format='JPEG')
+        img_byte_arr.seek(0)
+
+        # Path where the image will be temporarily stored
+        temp_image_path = os.path.join(os.getcwd(), 'new_profile_image.jpg')
+
+        # Saving the imaeg on path
+        with open(temp_image_path, 'wb') as f:
+            f.write(img_byte_arr.getvalue())
+
+        # Mocking the QFileDialog to select the image
+        with mock.patch('PyQt5.QtWidgets.QFileDialog.getOpenFileName', return_value=(temp_image_path, 'Image (*.jpg)')):
+            self.window.mscolab.upload_image()
+
+        qtbot.wait(1000)
```

Comment:
`qtbot.wait` should only be used as a last resort, since using it doesn't solve the underlying race condition, it just hides it (most of the time). It is better to use `qtbot.wait_until` and an assertion that ensures whatever is waited for has happened. `wait_until` would also only wait for as long as necessary, while a static duration with `wait` will be too long most of the time and slow the test suite down.